Exiled.Events/Patches/Events/Player/ActivatingWarheadPanel.cs

```diff
@@ -34,90 +31,82 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
-                new(OpCodes.Brfalse_S, eventLabel),
-
-                // if (Player.Get(this._hub).CurrentItem != null)
-                //    if (player.CurrentItem is Keycard keycard)
-                //        if (keycard.Permissions.HasFlag(AlphaWarhead))
-                //            isAllowed = true;
-                new(OpCodes.Ldloc_S, player.LocalIndex),
-                new(OpCodes.Callvirt, PropertyGetter(typeof(Player), nameof(Player.CurrentItem))),
-                new(OpCodes.Dup),
-                new(OpCodes.Stloc_S, curItem),
-                new(OpCodes.Brfalse_S, eventLabel),
-                new(OpCodes.Ldloc_S, curItem),
-                new(OpCodes.Isinst, typeof(Keycard)),
-                new(OpCodes.Dup),
-                new(OpCodes.Stloc_S, keycard.LocalIndex),
-                new(OpCodes.Brfalse_S, eventLabel),
-                new(OpCodes.Ldloc_S, keycard),
-                new(OpCodes.Callvirt, PropertyGetter(typeof(Keycard), nameof(Keycard.Permissions))),
-                new(OpCodes.Box, typeof(KeycardPermissions)),
-                new(OpCodes.Ldc_I4_8),
-                new(OpCodes.Box, typeof(KeycardPermissions)),
-                new(OpCodes.Call, Method(typeof(System.Enum), nameof(System.Enum.HasFlag))),
-                new(OpCodes.Brfalse_S, eventLabel),
-                new(OpCodes.Ldc_I4_1),
-                new(OpCodes.Stloc_S, isAllowed.LocalIndex),
-            });
-
-            offset = -2;
-            index = newInstructions.FindLastIndex(i => i.opcode == OpCodes.Ldloc_2) + offset;
-
-            newInstructions.InsertRange(index, new[]
-            {
-                // ev = new(player, isAllowed)
-                new CodeInstruction(OpCodes.Ldloc, player.LocalIndex).WithLabels(eventLabel),
-                new(OpCodes.Ldloc, isAllowed.LocalIndex),
-                new(OpCodes.Newobj, GetDeclaredConstructors(typeof(ActivatingWarheadPanelEventArgs))[0]),
-                new(OpCodes.Dup),
-
-                // Handlers.Player.OnActivatingWarheadPanel
-                new(OpCodes.Call, Method(typeof(Handlers.Player), nameof(Handlers.Player.OnActivatingWarheadPanel))),
-
-                // if (!ev.IsAllowed)
-                //    return;
-                new(OpCodes.Callvirt, PropertyGetter(typeof(ActivatingWarheadPanelEventArgs), nameof(ActivatingWarheadPanelEventArgs.IsAllowed))),
-                new(OpCodes.Brfalse_S, retLabel),
-            });
-
-            newInstructions[newInstructions.Count - 1].labels.Add(retLabel);
+            LocalBuilder player = generator.DeclareLocal(typeof(Player));
+            LocalBuilder allowed = generator.DeclareLocal(typeof(bool));
+            LocalBuilder card = generator.DeclareLocal(typeof(Keycard));
+
+            int index = newInstructions.FindIndex(i => i.Is(OpCodes.Ldfld, Field(typeof(PlayerInteract), nameof(PlayerInteract._sr))));
+
+            newInstructions.RemoveRange(index, 17);
+
+            newInstructions[index].labels.Add(retLabel);
```

Comment:
Rename it continueLabel or anything you want but not ret since it's will not going to a return direclty